### ISSUE TITLE
Inconsistent Behavior with transforms.v2 for Multiple Arguments

### ISSUE THREAD

**Issue by sanghunpark** [pytorch/vision#8573]
### 🐛 Describe the bug

I've been testing various transforms.v2 and noticed an inconsistency:

- When passing multiple PIL.Image arguments, the transformation is applied to all of them simultaneously, which is the expected behavior.
- However, when passing multiple torch.Tensor arguments, only the first argument is transformed, while the others remain unchanged. This issue isn't just with Resize; it also occurs with other transforms like Normalize.
Wouldn't it be more intuitive and consistent if transforms were applied to all provided arguments, whether they are PIL.Image or torch.Tensor? This would lead to a more predictable and natural workflow, especially when dealing with multiple tensors.

Expected Behavior:
- The transform should be applied to all provided arguments, regardless of their type (PIL.Image or torch.Tensor), in a consistent manner.

This inconsistency becomes especially confusing when using Compose to chain multiple transforms. The behavior differs depending on whether the input is a PIL.Image or a torch.Tensor, which can lead to unexpected results and confusion.

```
import torch
import torchvision.transforms.v2 as v2
import PIL.Image as Image


# PIL Image
a = Image.new("L", (2, 2), (1))
b, c = a.copy(), a.copy()

print('-- (pil.image) resize applied --')
print(a.size, b.size, c.size)
a, b, c = v2.Resize(1)(a, b, c)
print(' >> ')
print(a.size, b.size, c.size)


# torch
a = torch.ones([1, 2, 2])
b, c = a.clone(), a.clone()

print('-- (torch) resize applied --')
print(a.shape, b.shape, c.shape)
print(' >> ')
a, b, c = v2.Resize(1)(a, b, c)
print(a.shape, b.shape, c.shape)
```

```
-- (pil.image) resize applied --
(2, 2) (2, 2) (2, 2)
 >> 
(1, 1) (1, 1) (1, 1)
-- (torch) resize applied --
torch.Size([1, 2, 2]) torch.Size([1, 2, 2]) torch.Size([1, 2, 2])
 >> 
torch.Size([1, 1, 1]) torch.Size([1, 2, 2]) torch.Size([1, 2, 2])
```

OS: Ubuntu 20.04.6 LTS
PyTorch: 2.3.1
Torchvison: 0.18.1

### Versions

PyTorch version: 2.3.1
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.26.4
Libc version: glibc-2.35

Python version: 3.10.14 (main, May  6 2024, 19:42:50) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-97-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 4090
GPU 1: NVIDIA GeForce RTX 4090

Nvidia driver version: 535.161.07
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             32
On-line CPU(s) list:                0-31
Vendor ID:                          GenuineIntel
Model name:                         Intel(R) Core(TM) i9-14900K
CPU family:                         6
Model:                              183
Thread(s) per core:                 2
Core(s) per socket:                 24
Socket(s):                          1
Stepping:                           1
CPU max MHz:                        6000.0000
CPU min MHz:                        800.0000
BogoMIPS:                           6374.40
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req umip pku ospke waitpkg gfni vaes vpclmulqdq tme rdpid movdiri movdir64b fsrm md_clear serialize pconfig arch_lbr flush_l1d arch_capabilities
Virtualization:                     VT-x
L1d cache:                          896 KiB (24 instances)
L1i cache:                          1.3 MiB (24 instances)
L2 cache:                           32 MiB (12 instances)
L3 cache:                           36 MiB (1 instance)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-31
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] optree==0.11.0
[pip3] torch==2.3.1
[pip3] torchaudio==2.3.1
[pip3] torchelastic==0.2.2
[pip3] torchvision==0.18.1
[pip3] triton==2.3.1
[conda] blas                      1.0                         mkl  
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] libjpeg-turbo             2.0.0                h9bf148f_0    pytorch
[conda] mkl                       2023.1.0         h213fc3f_46344  
[conda] mkl-service               2.4.0           py310h5eee18b_1  
[conda] mkl_fft                   1.3.8           py310h5eee18b_0  
[conda] mkl_random                1.2.4           py310hdb19cb5_0  
[conda] numpy                     1.26.4          py310h5f9d8c6_0  
[conda] numpy-base                1.26.4          py310hb5e798b_0  
[conda] optree                    0.11.0                   pypi_0    pypi
[conda] pytorch                   2.3.1           py3.10_cuda11.8_cudnn8.7.0_0    pytorch
[conda] pytorch-cuda              11.8                 h7e8668a_5    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                2.3.1               py310_cu118    pytorch
[conda] torchelastic              0.2.2                    pypi_0    pypi
[conda] torchtriton               2.3.1                     py310    pytorch
[conda] torchvision               0.18.1              py310_cu118    pytorch

**Comment by NicolasHug** [comment]
Hi @sanghunpark , thanks for the report.

What you're observing is the result of a compromise between backward compatibility, and extending the v2 transforms to support new use-cases. We cannot treat all tensors as images, because that would mean tensor labels would be mis-interpreted as images, creating more issues.

This is documented in the note here: https://pytorch.org/vision/main/auto_examples/transforms/plot_transforms_getting_started.html#what-do-i-pass-as-input

If you need to transform multiple images, it's best to convert the tensors to the `tv_tensors.Image` class first.
